core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -552,7 +552,10 @@ class ControllerApis(val requestChannel: RequestChannel,
   def handleBrokerRegistration(request: RequestChannel.Request): Unit = {
     val registrationRequest = request.body[BrokerRegistrationRequest]
     authHelper.authorizeClusterOperation(request, CLUSTER_ACTION)
-
+    val controllerNode = controllerNodes.find(node => node.id == registrationRequest.brokerId)
+    if (controllerNode.isDefined) {
+      info(s"A broker is registering with ${KafkaConfig.NodeIdProp}=${controllerNode.get.id} which is also in use by a controller.")
+    }
```

Comment:
I would remove this log message. This is a valid configuration and I don't think it is providing any additional information.